xls/public/c_api.h

```diff
@@ -60,11 +60,34 @@ bool xls_convert_dslx_to_ir(const char* dslx, const char* path,
                             size_t additional_search_paths_count,
                             char** error_out, char** ir_out);
 
+// As above, but also takes `enable_warnings` and `disable_warnings` which
+// are arrays of warning names to enable or disable respectively against the
+// default warning set.
+//
+// Note: `warnings_out` should be freed via `xls_c_strs_free`.
+bool xls_convert_dslx_to_ir_with_warnings(
+    const char* dslx, const char* path, const char* module_name,
+    const char* dslx_stdlib_path, const char* additional_search_paths[],
+    size_t additional_search_paths_count, const char* enable_warnings[],
+    size_t enable_warnings_count, const char* disable_warnings[],
```

Comment:
The argument count seems to be getting pretty high, how would you feel about merging all the warnings into structs? It seems we only have opaque structs, but we don't make any ABI stability promises and something like the following seems like it would help make it reasonable to use the API, right?

```
struct string_view_span {
  const char** views;
  size_t size;
};

struct warnings {
  struct string_view_span enable_warnings;
  struct string_view_span disable_warnings;
};
```

I'm not sure what all the considerations in using the C API are- is struct packing more obnoxious than it's worth?

Reply:
I just wanted to make potential for the ABI considerations minimal (including passing structs by value) by making them flat. If we passed the struct instead by pointer we'd need a more complicated invocation sequence.

In an abstract sense these C APIs are supposed to be fairly unchanging in that we probably make one with a different name if we need different functionality. Probably we won't keep strict rules just to make sure the project can continue evolving without undue burden, but that's roughly the tradeoff space I've been trying to hit for the C APIs.

Reply:
Yeah, that makes sense.